src/rgw/rgw_rest_s3.cc

```diff
@@ -3088,10 +3216,20 @@ RGWOp *RGWHandler_REST_Service_S3::op_post()
 
 RGWOp *RGWHandler_REST_Bucket_S3::get_obj_op(bool get_data)
 {
+   int list_type = s->info.args.get_int("list_type", &list_type, 1);
```

Comment:
the parameter is "list-type" with a - not a _ (see https://docs.aws.amazon.com/AmazonS3/latest/API/v2-RESTBucketGET.html)

`s->info.args.get_int()` returns an error code, not the value it reads. it should be safe to ignore its errors, since they'll just default to `list_type=1`

so more like this:
```c++
  int list_type = 1;
  s->info.args.get_int("list-type", &list_type, 1);
```

Reply:
I think the third parameter is the default type. Should we explicitly assign list_type to 1?

Reply:
Except the '-', both are same right?